apiserver/facades/client/application/application_unit_test.go

```diff
@@ -2189,6 +2199,46 @@ func (s *ApplicationSuite) TestDeployCAASModelDefaultStorageClass(c *gc.C) {
 	c.Assert(result.Results[0].Error, gc.IsNil)
 }
 
+func (s *ApplicationSuite) TestDeployCAASModelAttachStorageNotSupport(c *gc.C) {
+	// Unset feature flag
+	os.Unsetenv(osenv.JujuFeatureFlagEnvKey)
+	featureflag.SetFlagsFromEnvironment(osenv.JujuFeatureFlagEnvKey)
```

Comment:
I'd just have this test in the cli for deploy.
See other comment.

Reply:
Just want to clarify, so you want to remove this test?

Reply:
Yeah, if we add the feature flag check to the CLI, we can drop the check in the facade. That way, the CLI will fail early if the flag is not set. It also means that you can bootstrap a controller without the flag and just turn it on client side to test. It does man that clients like libjuju might be able to use the feature without the flag set. But given this is all temporary, it's fine just to do the checks at the edge IMO.